lib/vault.py

```diff
@@ -185,51 +185,42 @@ def configure(self):
-                print("aws-ec2 auth backend already created.")
-            client.write('auth/aws-ec2/config/client', access_key = aws_creds["aws_access_key"],
-                                                       secret_key = aws_creds["aws_secret_key"])
-
-            arn_prefix = 'arn:aws:iam::{}:instance-profile/'.format(aws_creds["aws_account"])
-            policies = [p for p in provisioner_policies if p not in ('provisioner',)]
-            for policy in policies:
-                client.write('/auth/aws-ec2/role/' + policy, policies = policy,
-                                                             bound_iam_role_arn = arn_prefix + policy)
-
-        # AWS Secret Backend
-        if aws_creds is None:
-            print("Vault AWS credentials file does not exist, skipping configuration of AWS secret backend")
+                client.enable_auth_backend('aws')
+            except Exception as e:
+                print("Error while enabling auth back end: " + e)
+                sys.exit(1)
         else:
+            print("aws auth backend already created.")
+
+        #Define policies and arn                                     
+        policies = [p for p in provisioner_policies if p not in ('provisioner',)]
+        arn = 'arn:aws:iam::{}:instance-profile/'.format(os.environ['AWS_ACCOUNT'])
+        #TODO: Find a temporary way of storing the aws account number
+        #For each policy configure the policies on a role of the same name
+        for policy in policies:
+            client.write('/auth/aws/role/' + policy, auth_type='ec2', bound_iam_instance_profile_arn= arn + policy, policies=policy)
+            print('Successful write to aws/role/' + policy)
+        
+        # AWS Secret Backend
+        if 'aws' not in client.list_secret_backends():
             try:
                 client.enable_secret_backend('aws')
-            except hvac.exceptions.InvalidRequest as ex:
-                print("aws secret backend already created.")
-            client.write("aws/config/root", access_key = aws_creds["aws_access_key"],
-                                            secret_key = aws_creds["aws_secret_key"],
-                                            region = aws_creds.get("aws_region", "us-east-1"))
-            client.write("aws/config/lease", lease = aws_creds.get("lease_duration", "1h"),
-                                             lease_max = aws_creds.get("lease_max", "24h")) # DP TODO finalize default values
-
-            path = os.path.join(POLICY_DIR, "*.iam")
-            for iam in glob.glob(path):
-                name = os.path.basename(iam).split('.')[0]
-                with open(iam, 'r') as fh:
-                    # if we json parse the file first we can use the duplicate key trick for comments
-                    client.write("aws/roles/" + name, policy = fh.read())
+            except Exception as e:
+                print('Error while enabling secret back end: ' + e)
+                sys.exit(1)
```

Comment:
A Library should never call `sys.exit()` as it disrupts the caller. It should raise an exception or similar if it cannot complete the requested action.

Reply:
Done